library/src/main/kotlin/com/livefront/gsonkotlinadapter/KoltinReflectiveTypeAdapterFactory.kt

```diff
@@ -106,6 +104,8 @@ class KotlinReflectiveTypeAdapterFactory private constructor(
         }
 
         override fun read(reader: JsonReader): T? {
+            require(!kClass.isAbstract) { "Cannot deserialize abstract class '${kClass.simpleName}'" }
+            require(!kClass.isSealed) { "Cannot deserialize sealed class '${kClass.simpleName}'" }
```

Comment:
👍

Reply:
So I know this forbids reading to `SealedParent` but does it allow reading to `SealedParent.Child`?

Reply:
It does not, assuming the child is not a sealed class itself.


```kotlin
sealed class Foo {
    data class Bar(val integer: Int): Foo()
}
```
I did a little testing and `isSealed` returns true for `Foo`, but not for `Bar`. Similar story with a abstrrac class.

Reply:
So does reading to `Foo.Bar` work or not though? I know `Bar` won't trigger `isSealed` but I'm just wondering if at some point there is a lookup to the parent class when serialization.

Reply:
It will work

Reply:
OK well that's good to know then.